### PR TITLE
Fix RedundantPresenceValidationOnBelongs on some files (part II)

### PR DESCRIPTION
**Pull request by cyrillefr**
#### What? Why?
- Contributes to #11482
- First attempt was made by #12380 , but some choices will have to be made file by file. So, here 5 files deemed the simpler ones (cf. conversation). Hopefully the rest will follow.
-  Continuation of #12407 
- See also #11297 that started the work

#### What should we test?
After migration, all tests should pass and no rubocop offense should be raised.
Run the 3 rake files, to check it any corrupt data is present.

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [X] Technical changes only
- [ ] Feature toggled

The title of the pull request will be included in the release notes.

### CODE REVIEW

File: app/models/spree/product_property.rb [hunk 1]
```diff
@@ -2,12 +2,9 @@
 
 module Spree
   class ProductProperty < ApplicationRecord
-    self.belongs_to_required_by_default = false
-
-    belongs_to :product, class_name: "Spree::Product", touch: true
+    belongs_to :product, class_name: "Spree::Product", touch: true, optional: true
```
Comment (rioug) [inline app/models/spree/product_property.rb:5]: It don't think it makes sense to have the product optional here. 
I checked the production server and none of them have a product property with a `product_id` set to null so it's safe to the change. @openfoodfoundation/core-devs  what do you think ?

Comment (cyrillefr) [reply app/models/spree/product_property.rb:5]: Well if there are no null `product_id`, I guess I could also update the migration file.

Comment (dacook) [reply app/models/spree/product_property.rb:5]: Indeed, ProductProperty is a join table between Product and Property, so the association is definitely required.

So both the product_id and property_id columns can become `NOT NULL`. I'm not sure if Gaetan checked all prod servers yet, so we might need to check that before deploying.

**Review by dacook — CHANGES_REQUESTED** [review]
Thanks. We would like to make `spree_product_properties.product_id` required as well.

File: app/models/spree/product_property.rb (hunk 1 above)
Comment (cyrillefr) [reply app/models/spree/product_property.rb]: Cool. So I have made the update.

**Review by dacook — APPROVED** [review]
Great, thank you!

**Review by rioug — APPROVED** [review]
Nice, thank you

**Comment by dacook** [comment]
Migration succeeded on au_staging:

```
deploy : migrate database ---------------------------------------------- 10.56s
```

@rioug just double-checking, can you confirm we are good to merge?

**Comment by rioug** [comment]
We are all good, merging.

**Pull request merged**